File: src/main/java/jukeboxmc/block/BlockFlowerPot.java
package jukeboxmc.block;

import org.jukeboxmc.item.ItemFlowerPot;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockFlowerPot extends BlockWaterlogable {

    public BlockFlowerPot() {
        super( "minecraft:flower_pot" );
    }

    @Override
    public ItemFlowerPot toItem() {
        return new ItemFlowerPot();
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.FLOWER_POT;
    }

    @Override
    public boolean isSolid() {
        return false;
    }

    @Override
    public boolean isTransparent() {
        return true;
    }

    public void setUpdate( boolean value ) {
        this.setState( "update_bit", value ? (byte) 1 : (byte) 0 );
    }

    public boolean isUpdate() {
        return this.stateExists( "update_bit" ) && this.getByteState( "update_bit" ) == 1;
    }

}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockFlowerPot.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.FlowerPot
import org.jukeboxmc.api.block.data.BlockFace
import org.jukeboxmc.api.blockentity.BlockEntity
import org.jukeboxmc.api.blockentity.BlockEntityType
import org.jukeboxmc.api.math.Vector
import org.jukeboxmc.server.block.JukeboxBlock
import org.jukeboxmc.server.extensions.toByte
import org.jukeboxmc.server.item.JukeboxItem
import org.jukeboxmc.server.player.JukeboxPlayer
import org.jukeboxmc.server.world.JukeboxWorld

class BlockFlowerPot(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates), FlowerPot {

    override fun placeBlock(
        player: JukeboxPlayer,
        world: JukeboxWorld,
        blockPosition: Vector,
        placePosition: Vector,
        clickedPosition: Vector,
        itemInHand: JukeboxItem,
        blockFace: BlockFace
    ): Boolean {
        val block = world.getBlock(placePosition)
        if (block is BlockWater && block.getLiquidDepth() == 0) {
            world.setBlock(placePosition, block, 1, false)
        }
        BlockEntity.create(BlockEntityType.FLOWERPOT, this)?.spawn()
        return super.placeBlock(player, world, blockPosition, placePosition, clickedPosition, itemInHand, blockFace)
    }

    override fun isUpdate(): Boolean {
        return this.getBooleanState("update_bit")
    }

    override fun setUpdate(value: Boolean): FlowerPot {
        return this.setState("update_bit", value.toByte())
    }

    override fun getWaterLoggingLevel(): Int {
        return 1
    }
}